### ISSUE TITLE
Is this ready for use?

### ISSUE THREAD

**Issue by spences10** [rorz/react-typist-cycle#1]
I just removed `react-typist` and replaced with this, as I'm guessing that there is a dependency for `react-typist` within `react-typist-cycle` correct?

I'm asking as I got an unexpected token in the `App.js` of `react-typist-cycle` apologies I did not take screenshots or details, if you need it recreated I'll do so

I'm using windows 10 with chrome canary if that makes any difference

thanks

**Comment by rorz** [comment]
Hi, @spences10 

**tl;dr — You're right that it is not ready for general use right now.**

Thanks for trying this out! I am only just trying to create my own React packages and feel that I have missed a few things on this (hence why I removed the link to its npm repo for now).

There is indeed a dependency for react-typist (should show up in the project.json). However, because of how _bare-bones_ the project is at the moment (i.e. just a .jsx file) it is likely to break in most environments. It theoretically _should_ work in environments that have their own babel and webpack loaders, such as [create-react-app](https://github.com/facebookincubator/create-react-app) and [gatsbyJS](https://github.com/gatsbyjs).

Do you mind me asking what environment you are using this in? (i.e. are you using something like create-react-app?)

I am working ASAP to get the project re-packaged into a self-contained dependency that is environment agnostic. Your OS/browser shouldn't be the issue at the moment.

**Comment by spences10** [comment]
hi @rorz,

Yeahp using `create-react-app` with my own `ESLint` settings, I'm just getting used to react so am a learner really doing all the simple projects I can, so I wanted to use this in a markdown previewer.

Basically I want to use either `react-typist`  or `react-typist-cycle` to enter text onto the `<textarea>` but found that neither worked, can either of these type into a text area?

**Comment by rorz** [comment]
ESLint shouldn't interfere with the workings of the code; it should work out of the box with `create-react-app`.

That's a very interesting use case! Can I ask why you want to do that? I am curious because I can't think how a user would be able to edit dynamically-created content.

You can't put anything but raw text into a HTML `<textarea>` element. However, you can create a `<div>` that behaves much the same way, as described in [this SO answer](http://stackoverflow.com/a/21528880/1180964).

A React-ready version of the above would be:
```jsx
<div contentEditable="true" style={{ minHeight: '50px', width: '300px' }}>
  // Put whatever you want here, i.e. <TypistCycle ... />
</div>
```

**However, ** having tried it with a `<TypistCycle ... />` element just now, I can confirm that the animation stops as soon as you try to edit the content... Perhaps this is what you want? I am unable to look into supporting this right now (as it will relate directly to the `react-typist` project).

**Comment by spences10** [comment]
hi @rorz, yes that's exactly the behaviour I want, so [in my mind]

1. page loads
2. text area has some text added to it so the user can see the markdown in action
3. they click in the `<textarea>` and start adding their own content 

So, I've just had the briefest of looks at the SO question and can see you have added the main part of the code here, I'm going to try this now with `react-typist` and wait, patiently for `react-typist-cycle` to get off the ground

Thanks for the pointers :+1:  I'm watching this repo now for more updates :100:

**Comment by callil** [comment]
@spences10 Here's how to use this project with create-react-app

1. Download this repository. 
2. Copy `index.jsx` to `/src` directory in your project
3. Rename the file to `typeCycle.jsx`
4. Include the file in your App.js by writing `import TypistCycle from './typeCycle';`
5. Copy the sample code from this github into your render method

if you want a gist to see the code first hand let me know!

**Comment by rorz** [comment]
Thanks for checking this out, @callil ! — And for providing installation info.... Do you want to provide this as a PR to the README? If not, I'll update it.

And apologies to @spences10 ... I had neglected this for a while but I'll add it to my radar as I think I'm still getting traffic from NPM where it won't function as a direct installation...

**Comment by callil** [comment]
Sure I'll make a PR if it's helpful. I'm getting some other small issues in console (don't seem to affect performance) that I'm a bit stumped on so I might open another issue on those :)

**Comment by spences10** [comment]
>if you want a gist to see the code first hand let me know!

Sure @callil, that would be great, thank you 👍 

I'll close this and reference it in the project I was going to use it in 

Thanks
